### PR TITLE
Windows: change kerberos service name to postgres

### PR DESCRIPTION
**Pull request by pf-qiu**
The kerberos service name of gpdb and psql must match to allow proper
kerberos authentication. For linux platform this is controlled by
--with-krb-srvnam. For Windows platform it's hardcoded in template
pg_config.h.win32. We currently build gpdb server without specifying
--with-krb-srvnam, and the default value is "postgres". Change the
correspond value that hardcoded Windows config template to "postgres"
also.

### CODE REVIEW

File: concourse/scripts/ic_gpdb_remote_windows.bat [hunk 1]
```diff
@@ -2,10 +2,17 @@ set PGPORT=%1
...
 set PGHOST=127.0.0.1
 
+mkdir C:\ProgramData\MIT\Kerberos5\
+move krb5.ini C:\ProgramData\MIT\Kerberos5\
+set KRB5CCNAME=%USERPROFILE%\krb5cache
+set PGGSSLIB=gssapi
+
 call "C:\Program Files\Greenplum\greenplum-clients\greenplum_clients_path.bat"
+kinit -k -t gpdb-krb5.keytab user1/127.0.0.1
 set path=%path%;C:\Program Files\curl-win64-mingw\bin
```
Comment (fanfuxiaoran) [inline concourse/scripts/ic_gpdb_remote_windows.bat]: Maybe better to call klist here. If there something goes wrong, we can get some information from klist.

File: concourse/scripts/ic_gpdb_remote_windows.bash [hunk 2]
```diff
@@ -115,6 +171,9 @@ function _main() {
         tar xzvf *.tar.gz
     popd
 
+    time install_packages
+    time setup_gpadmin_user
+    time setup_kerberos
     time create_cluster
```
Comment (fanfuxiaoran) [inline concourse/scripts/ic_gpdb_remote_windows.bash]: Can put `setup_kerberos` after create_cluster. In `create_cluster` function `setup_gpadmin` is called.

File: concourse/scripts/ic_gpdb_remote_windows.bash [hunk 3]
```diff
@@ -22,9 +24,59 @@ function configure_gpdb_ssl() {
...
+
+# Start KDC on master node
+kdb5_util create -s -r ${DEFAULT_REALM} -P changeme
+
+kadmin.local -q "addprinc -pw changeme root/admin"
+
+krb5kdc
+
+kadmin.local -q "addprinc -randkey user1/127.0.0.1@${DEFAULT_REALM}"
+kadmin.local -q "addprinc -randkey user2/127.0.0.1@${DEFAULT_REALM}"
```
Comment (fanfuxiaoran) [inline concourse/scripts/ic_gpdb_remote_windows.bash]: This line is useless.

File: concourse/scripts/ic_gpdb_remote_windows.bash [hunk 4]
```diff
@@ -22,9 +24,59 @@ function configure_gpdb_ssl() {
...
+kadmin.local -q "addprinc -randkey postgres/127.0.0.1@${DEFAULT_REALM}"
+
+# Gernerate keys
+rm -rf /home/gpadmin/gpdb-krb5.keytab
+kadmin.local -q "xst -norandkey -k /home/gpadmin/gpdb-krb5.keytab user1/127.0.0.1@${DEFAULT_REALM}"
```
Comment (fanfuxiaoran) [inline concourse/scripts/ic_gpdb_remote_windows.bash]: This line is useless.

Comment (pf-qiu) [reply concourse/scripts/ic_gpdb_remote_windows.bash]: Both server and client needs a keytab, it's shared here. But we should separate them.

File: concourse/scripts/ic_gpdb_remote_windows.bash (hunk 2 above)
Comment (pf-qiu) [reply concourse/scripts/ic_gpdb_remote_windows.bash]: setup_gpadmin_user is moved out here. It's a prerequisite for many function, not a good idea to hide it inside one.

**Pull request merged**